cmd/tiller-proxy/internal/handler/handler.go

```diff
@@ -59,6 +60,8 @@ func AuthGate() negroni.HandlerFunc {
 			return
 		}
 		ctx := context.WithValue(req.Context(), userKey, userAuth)
+		safe.Add(1)
```

Comment:
Just read Gorilla/mux graceful shudown section https://github.com/gorilla/mux#graceful-shutdown and they mentioned src.Shutdown which

> // Doesn't block if no connections, but will otherwise wait until the timeout deadline.

If that's the case, wouldn't that be enough instead of we maintaining a WaitGroup?

Reply:
I didn't find that. Yes, I guess it's better to use their recommended way.